clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -136,6 +136,12 @@ public void updateState(ConsumerGroupHeartbeatResponseData response) {
         }
     }
 
+    @Override
+    public boolean notInGroup() {
```

Comment:
I wonder if we could just call this "shouldHeartBeat" as the server side protocol is tight to the heartbeat

Reply:
Sure, sounds good to me